### PR TITLE
Update LFric container Xios, Yaxt, Tau, Psyclone 3

### PR DESCRIPTION
**Pull request by ScottWales**
Update the lfric container to the latest version being used for Rose Stem at NCI

https://code.metoffice.gov.uk/trac/lfric_apps/ticket/515

### CODE REVIEW

File: conda/psyclone/meta.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+{% set version = "3.0.0" %}
 
 package:
   name: "{{ name|lower }}"
   version: "{{ version }}"
 
 source:
-  url: https://github.com/stfc/PSyclone/archive/refs/tags/{{version}}.tar.gz
-  sha256: 00f5d4bed877293bd6327e1e545532a01ef288687f2e88b4f53799191bb819c3
+  url: https://github.com/stfc/PSyclone/archive/refs/tags/v{{version}}.tar.gz
```
Comment (hiker) [inline conda/psyclone/meta.yaml:9]: While I don't care, I seem to remember that spack prefers pypi package url over github.  If you think it makes sense to change, details including sha256 are here:
https://pypi.org/project/PSyclone/3.0.0/#psyclone-3.0.0.tar.gz

File: conda/psyclone/meta.yaml [hunk 2]
```diff
@@ -32,34 +32,12 @@ requirements:
     - python
     - six
     - sympy
+    - python-graphviz
```
Comment (hiker) [inline conda/psyclone/meta.yaml:35]: Admittedly we forgot to mention it in the release notes, but jinja2 is now also a required dependency.

Comment (hiker) [reply conda/psyclone/meta.yaml:35]: Also, can you add that fparser should be >=0.2.0?

File: environments/lfric/conda.yaml [hunk 3]
```diff
@@ -2,8 +2,9 @@ channels:
   - conda-forge
   - local
 dependencies:
-  - python
+  - python=3.11
   - psyclone
   - lfric-gpl-utils=2
   - jinja2
   - sci-fab
+  - python-clang
```
Comment (hiker) [inline environments/lfric/conda.yaml:10]: Nice, thanks!

File: environments/lfric/spack.yaml [hunk 4]
```diff
@@ -1,8 +1,10 @@
 spack:
   specs:
-    - xios@2.5.2252 +lfric
-    - yaxt@0.9.0 idxtype=long
+    - xios@2.5.2629 +lfric
+    - yaxt@0.11.3 idxtype=long
     - pfunit@3 max_array_rank=6 +mpi +openmp
+    - tau +mpi +openmp +opari ~pthreads +lfric
```
Comment (hiker) [inline environments/lfric/spack.yaml:6]: I will try that soon :)

**Review by hiker — CHANGES_REQUESTED** [review]
That does a lot more than just update the psyclone version :) I've added some comments, and really appreciate adding tau. I don't have the time to test this, but my understanding is that you/Dave verified it. Back to you to see if it's worth addressing the two or three minor issues, otherwise a great step forward :)

File: environments/lfric/spack.yaml (hunk 4 above)
Comment (ScottWales) [reply environments/lfric/spack.yaml:6]: This is a straight copy of what Jason has in the internal container, so should work just the same

**Comment by ScottWales** [comment]
Thanks Joerg,

I think using the github download url is fine, I've addressed the dependency changes

**Review by hiker — APPROVED** [review]
Thanks for updating the dependencies.

**Pull request merged**